Clarify documentation on series endpoint (#2953)
cortexproject/cortex: docs/configuration/v1-guarantees.md

Cortex strives to be 100% API compatible with Prometheus (under `/api/prom/*`); any deviation from this is considered a bug, except:

> - Requiring the `__name__` label on queries (except for queries to Ingesters).

Reviewer: ```suggestion
- Requiring the `__name__` label on queries when querying the chunks storage (queries to ingesters or clusters running the blocks storage are not affected).
```

Author: Thanks, used the suggested change.

> - For queries to the `/series` endpoint given time ranges older than the data stored in Ingesters will be ignored.

Reviewer: Few comments:
- I would name the API `/api/v1/series`
- This also affects `/api/v1/labels` and `/api/v1/label/{name}/values`
- Maybe it's more clear if we say that the API endpoints only query data from ingesters (not long-term storage)

Author: I have clarified the API path and added the mentioned endpoints. Also added that data is only retrieved from ingesters, wdyt?

Reviewer: LGTM, thanks!

> - For queries to the `/api/v1/series`, `/api/v1/labels` and `/api/v1/label/{name}/values` endpoints, given time ranges older than the data stored in ingesters will be ignored and data will only retrieved from ingesters.

Reviewer: ```suggestion
- For queries to the `/api/v1/series`, `/api/v1/labels` and `/api/v1/label/{name}/values` endpoints, query's time range is ignored and the data is always fetched from ingesters.
```
